errors.go

```diff
@@ -1,7 +1,65 @@
 package txfile
 
-import "errors"
+import (
+	"errors"
 
+	"github.com/elastic/go-txfile/internal/vfs"
+	"github.com/elastic/go-txfile/txerr"
+)
+
+type reason interface {
+	txerr.ErrorBuild
+}
+
+type ErrKind int
+
+// txfile internal error kinds
+var (
+// InvalidConfig = txerr.NewKind("configuration error")
+)
+
+// file system error kinds (map internal/vfs errors)
+const (
+	PermissionError       = vfs.ErrPermissions
+	FileExists            = vfs.ErrExist
+	FileDoesNotExist      = vfs.ErrNotExist
+	FileClosed            = vfs.ErrClosed
+	NoDiskSpace           = vfs.ErrNoSpace
+	FDLimit               = vfs.ErrFDLimit
+	CantResolvePath       = vfs.ErrResolvePath
+	IOError               = vfs.ErrIO
+	OSOtherError          = vfs.ErrOSOther
+	OperationNotSupported = vfs.ErrNotSupported
+	LockFailed            = vfs.ErrLockFailed
+)
```

Comment:
I did not name them ErrX, cause I see the `kind` more like a code or descriptive context element used by applications in order to classify the error (originally I have had another custom interface type). e.g. one will use these kinds like `if txerr.Is(txfile.PermissionError, err) { ... }` or `if txerr.Is(txfile.OutOfMemory, err)  { ... }`.

The vfs package uses `ErrX`... still some confusion on how to do this on my side.

Note, an error kind is of type `error`, because common interface of errors should be based on builtin types only (txerr should not enforce any other types). By forcing the `kind/class/code` to be type of type `error`, we enforces dev to implement `Error() string`. My original definition of kind was `type Kind interface { Describe() string }`. But this would require other packages to use the `txerr` package. Requiring `interface{}` would make the kind comparable as well, but would not enforce any of description.